src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/MemoryPoolViewBufferScope.cs

```diff
@@ -60,7 +65,7 @@ public ViewBufferValue[] GetSegment()
 
             try
             {
-                segment = _viewBufferPool.Rent(SegmentSize);
+                segment = _viewBufferPool.Rent(Math.Max(size, MinimumSize));
```

Comment:
What's the suggestion? I'm not going to make the code less correct than it already is because of `.Internal`